usecases/event-driven-pipeline/README.md

```diff
@@ -0,0 +1,311 @@
+
+## Prerequisites:
+
+Ensure that you have the following tools installed locally:
+
+1. [aws cli](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+2. [kubectl](https://Kubernetes.io/docs/tasks/tools/)
+3. [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli)
+
+## Deploy
+
+To provision this example:
+
+```bash
+git clone https://github.com/aws-controllers-k8s/examples.git
+cd usecases/event-driven-pipeline
+
+region=<your region> # set region variable for following commands
+terraform init
+terraform apply -var region=$region #defaults to us-west-2
+```
+
+Enter `yes` at command prompt to apply
+
+The following components are provisioned in your environment:
+- A sample VPC, 3 Private Subnets and 3 Public Subnets
+- Internet gateway for Public Subnets and NAT Gateway for Private Subnets
+- EKS Cluster Control plane with one managed node group
+- EKS Managed Add-ons: VPC_CNI, CoreDNS, Kube_Proxy, EBS_CSI_Driver
+- K8S cluster autoscaler and fluentbit agent
+- IAM execution roles for EMR on EKS, Step Functions and EventBridge 
+
+![terraform-output](img/terraform-output-eb-sfn-ack.png)
+
+### Validate
+
+The following command will update the `kubeconfig` on your local machine and allow you to interact with your EKS Cluster using `kubectl` to validate the deployment.
+
+### Run `update-kubeconfig` command:
+
+```bash
+aws eks --region us-west-2 update-kubeconfig --name event-driven-pipeline-demo
+```
+
+### List the nodes
+
+```bash
+kubectl get nodes
+
+# Output should look like below
+NAME                                        STATUS   ROLES    AGE     VERSION
+ip-10-1-10-64.us-west-2.compute.internal    Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-10-65.us-west-2.compute.internal    Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-10-7.us-west-2.compute.internal     Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-10-73.us-west-2.compute.internal    Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-11-96.us-west-2.compute.internal    Ready    <none>   19h     v1.24.9-eks-49d8fe8
+ip-10-1-12-197.us-west-2.compute.internal   Ready    <none>   19h     v1.24.9-eks-49d8fe8
```

Comment:
This is a lot of nodes? Is it worth scaling it down a little bit by default?

Reply:
we have two node groups in the cluster and each has 3 nodes for HA. Spark node group only for spark workloads and Core node group for controllers and all the other stuff. cluster_autoscaler  is installed for scaling